godot_project/project_workspace/workspace_context/workspace_context.gd

```diff
@@ -1202,14 +1202,33 @@ func snapshot_moment(in_operation_name: String) -> void:
 	# workaround for part of the state for this workspace_context being inside ScriptUtils (this will apply that state)
 	get_editable_structure_contexts()
 	
-	# Apply simulation if the operation modified the structure of the project
-	if not History.is_operation_whitelisted(in_operation_name):
-		apply_simulation_if_running()
-	
 	# apply overdue signals, ensure snapshots contain up to date state
 	_emit_selection_in_structures_changed()
 	_emit_structure_content_changed()
 	
+	# Apply simulation if the operation modified the structure of the project
+	if not History.is_operation_whitelisted(in_operation_name) and is_simulating():
```

Comment:
unrelated to this PR, but I dont like the function name `History.is_operation_whitelisted()` it lacks contexts imho. What means to be whitelisted? it made more sense being part of SimulationStateCache because that class was about simulation, but History is an overall thing.

Reply:
Renamed it to "is_operation_whitelisted_during_simulation", not perfect but maybe we can move the list itself outside the of the History class